src/app/utils/field-validation/field-validation.guards.ts

```diff
@@ -0,0 +1,49 @@
+import { types as basicTypes } from '../../../shared/resources/basic'
+import { BasicField } from '../../../types/field'
+import {
+  ProcessedCheckboxResponse,
+  ProcessedFieldResponse,
+  ProcessedSingleAnswerResponse,
+  ProcessedTableResponse,
+} from '../../modules/submission/submission.types'
+
+const singleAnswerFieldTypes = basicTypes
+  .filter((field) => !field.answerArray)
+  .map((f) => f.name)
+
+export const isProcessedSingleAnswerResponse = (
+  response: ProcessedFieldResponse,
+): response is ProcessedSingleAnswerResponse => {
+  return (
+    'answer' in response &&
+    typeof response.answer === 'string' &&
+    singleAnswerFieldTypes.includes(response.fieldType)
+  )
+}
+
+const isStringArray = (arr: unknown): arr is string[] =>
+  Array.isArray(arr) && arr.every((item) => typeof item === 'string')
+
+const isNestedStringArray = (arr: unknown): arr is string[][] => {
+  return Array.isArray(arr) && arr.every(isStringArray)
+}
+
+export const isProcessedCheckboxResponse = (
+  response: ProcessedFieldResponse,
+): response is ProcessedCheckboxResponse => {
+  return (
+    'answerArray' in response &&
+    isStringArray(response.answerArray) &&
+    response.fieldType === BasicField.Checkbox
```

Comment:
nit: should the fieldType check come first? so the other checks are not for naught.